src/payment_store.rs

```diff
@@ -153,6 +157,14 @@ where
 	pub(crate) fn insert(&self, payment: PaymentDetails) -> Result<bool, Error> {
 		let mut locked_payments = self.payments.lock().unwrap();
 
+		let payment = PaymentDetails {
```

Comment:
why not set `payment.last_update` directly?

Reply:
`payment` is immutable in this context. We could do that, or redeclare it as `mut` — it's a matter of preference I guess.

Reply:
ah ok, it sounds fine as-is. Thanks!